### ISSUE TITLE
RDCP objects JPG views in curator mode get logged as non-curator views

### ISSUE THREAD

**Issue by hjsyoo** [ucsdlib/damsmanager#218]
### Descriptive summary

This testing was conducted on RDCP objects which contain one or more JPG file components. In curator mode, when we click on the JPG files in an RDCP object to view the pictures, the JPG views do not register as curator views. Instead, they register as non-curator views.

### Rationale

When curators visit the object page to view/manage the JPG file components, their visits will be recorded as non-curator visits, which results in inaccurate reporting.

### Steps to reproduce the behavior

- Our Testing Procedure: (This testing is done through Chrome under UCSD IP.)
1.  Log in to curator mode.
2. Open the object page by directly clicking on the URL, which has been saved on a google doc/sheet (this particular view won’t be logged, as we learned from testing direct links from google sheets).
3. Click on the object’s components which contain JPG files. 
4. Click on the image itself to view the pictures.
5. Click “backward” arrow button in the browser to redirect back to the object page.
6. If there is more than one image, repeat step 2-6 until the last JPG file in the object is opened and viewed.
- Testing Documentation:
-- Google Spreadsheet links. Testing JPG “View File”: https://docs.google.com/spreadsheets/d/1JWic6IF1e3yswqqNmOxxFy35lIwuhu7_Kbuvz62ynbs/edit#gid=643112126
-- Under the sixth tab/sheet of the google spreadsheet (Testing JPG “View File”), line 5 to line 22 records the data from testing under non-curator mode, line 26 to line 43 records the data from testing under curator mode.
-- Taking line 26 in the “Testing URL direct links” sheet as an example, we went through the testing procedure under curator mode and viewed both of the JPG files that the object contains. Since we opened the object page by directly clicking on the URL, no view should be logged for the very first time we were in the object page (from the previous testing on “Testing URL direct links”, we conclude that no views are logged when we open an object by directly clicking on its URL). Therefore, we expect the curator view counts to be the number of times we viewed the JPG files.  Cell H26 indicates that we would expect two separate views being logged under curator views. Cell J26 indicates that in the DAMs report, none of those views were counted toward curator views. Cell I26 indicates that those expected views were counted toward non-curator views.
- Clarification on Spreadsheet’s Header
-- Expected Views: The number of times we click on the JPG image to view the pictures. 
-- Observed Views: The data difference between two consecutive days’ view counts in CSV report. (Note: CSV report totals up the new view counts with the previous view counts for that month at midnight.)
-- Note: Column L to Column P are the replicate testing we did to make sure this phenomenon is not accidental.

### Related work

- RDCP object views from external points of entry do not get logged at all #225
- RDCP objects JPG views in curator mode get logged as non-curator views #218
- RDCP object views from external points of entry in curator mode get logged as non-curator views #226
- RDCP object views get logged inconsistently from session to session #227
- RDCP object views do not get logged uniquely #219

**Comment by hjsyoo** [comment]
@gamontoya @jessicahilt My team needs to do additional testing on the RDCP stats reports before we can write up complete specs on a third related ticket - we discovered inconsistent results at the end of last week. Since we're all out of office until April, is it possible to put this work on hold until further notice?

**Comment by gamontoya** [comment]
`On hold until further notice.`

**Comment by lsitu** [comment]
@hjsyoo I think this ticket is related to my comment https://github.com/ucsdlib/damsmanager/issues/219#issuecomment-378395162 as well. When clicking on the image viewer, it's redirected to zoomify, which won't be counted. So the view of the object will be counted for the object URL in the google spreadsheet, which is public. Please let me know if I miss anything when conducting any more tests. Thanks.

**Comment by lsitu** [comment]
@hjsyoo As specifying in `Steps to reproduce the behavior` :
```
2. Open the object page by directly clicking on the URL, which has been saved on a google doc/sheet (this particular view won’t be logged, as we learned from testing direct links from google sheets).
```
It looks like that test here is over wrapping with ticket https://github.com/ucsdlib/damsmanager/issues/226 (RDCP object views from external points of entry in curator mode get logged as non-curator views). I have to clarify that if 
an object view is accessed from a external URL saved on a google doc/sheet, it will be always counted as one non-curator view. And with the current  implementation to extract data from the weblog, we are not able to count it as curator view.

However, with the subsequent actions in step # 3 and # 4 to click on the jpeg image , it's opening the zoomify viewer that we haven't counted on yet:
```
3. Click on the object’s components which contain JPG files.
4. Click on the jpeg image itself to view the pictures.
```

So I think the issue here is how we are going to count this sequence of actions:
- Click on the external link saved on a google doc/sheet, which opens the object viewer.
- Click on the jpeg image in the object viewer, which opens the zoomify viewer.

When `Click on the external link saved on a google doc/sheet, which opens the object viewer`, we've counted it as a non-curator view as what you've seen now.

When `Click on the jpeg image in the object viewer, which opens the zoomify viewer`, do you want to count it as a view of the object as well? And how are you going to count it when there are more than one component with jpeg images in an object?

**Comment by hjsyoo** [comment]
@lsitu Thanks, your understanding of the issues reported is correct, but I think we should put a hold on all of this stats work until @gamontoya @mcritchlow @jessicahilt have had a chance to meet next week to discuss these 5 related tickets. I'm not sure at this point if actual work should be done, or if the tickets should be considered guidelines for testing stats reporting in D5.
But to answer your last question about the expected outcome of clicking on the jpeg image in object viewer, we didn't have an expectation, per se. We just found that those clicks did register as views, but curator clicks registered as non-curator views. This ticket is to report this finding.

**Comment by lsitu** [comment]
@hjsyoo Okay, I'll hold off those five dams stats related tickets for now.
